doc/examples/reconst_ivim.py

```diff
@@ -0,0 +1,193 @@
+physiologic meaning, one expects that D* is greater than D.
+
+The IVIM model expresses the diffusion signal as follows:
+
+ .. math: :
+
+    S(b) = S_{0}[fe ^ {-b*D_star} + (1 - f)e^{-b*D}]
+
+where $\mathbf{b}$ is the gradient value (which is dependent on
+the measurement parameters), $S_0$ is the signal in the absence
+of diffusion gradient sensitization, $\mathbf{f}$ is the perfusion
+fraction, $\mathbf{D}$ is the diffusion coefficient,
+ $\mathbf{D * }$ is the pseudo diffusion constant.
+
+In the following example we show how to fit the IVIM model
+on diffusion  weighted dataset and visualize the perfusion
+and diffusion. First, we import all relevant modules:
+"""
+
+import numpy as np
+from scipy.optimize import leastsq
+import matplotlib.pyplot as plt
+
+from dipy.reconst.ivim import IvimModel, _ivim_error, ivim_function
+from dipy.data.fetcher import read_ivim
+
+"""
+We get an IVIM dataset using dipy's data fetcher read_ivim.
+The image data is an acquired IVIM dataset with 21 b - values
+acquired in 3 directions, registered to each other and the
+directions were averaged into one. So, this dataset is 4D
+where the 4th dimension is the b - values. The fetcher
+read_ivim gets the .nii, .bval and .bvec files.
+"""
+
+img, gtab = read_ivim()
+
+"""
+img contains a nibabel Nifti1Image object(with the data)
+and gtab contains a GradientTable object(information about
+the gradients e.g. b - values and b - vectors). We get the data from img using read_data.
+"""
+
+data = img.get_data()
+print('data.shape (%d, %d, %d, %d)' % data.shape)
+
+"""
+Thus the data is a 256x256x54 volume and the fourth dimension
+specifies the bvalue. Let us visualize the data by taking a
+slice midway(z=27) at bvalue = 0.
+"""
+
+z = 33
+b = 20
+
+plt.imshow(data[:, :, z, b].T, origin='lower', cmap='gray')
+plt.axhline(y=100)
+plt.axvline(x=170)
+plt.savefig("brain_slice.png")
+plt.close()
+
+"""
+The marked point in the image shows a section containing cerebral spinal fluid (CSF)
+so it should have a very high f and D*, the area between the right and left is white
+matter so that should be lower, and the region on the right is gray matter and CSF.
+That should give us some contrast to see the values varying across the regions.
+Let us now take a section of the slice for fitting the IVIM model.
+"""
+
+x1, x2 = 160, 180
+y1, y2 = 90, 110
+
+plt.imshow(data[x1:x2, y1:y2, z, b].T, origin='lower', cmap="gray")
+plt.savefig("CSF_slice.png")
+plt.close()
+
+"""
+Now that we have prepared the datasets we can go forward with
+the ivim fit. Instead of fitting the entire volume, we will fit
+for the slice of data containing CSF. (x1, x2), (y1, y2) and z
+specify the coordindates of the section.
+"""
+
+data_slice = data[x1:x2, y1:y2, z, :]
+
+"""
+First, we instantiate the Ivim model in the following way.
+"""
+
+ivimmodel = IvimModel(gtab)
+
+"""
+Fitting the data is very simple. We just need to call the fit method of the
```

Comment:
I don't think we need the first sentence. Instead, how about: "To fit the model, call the `fit` method, specifying the `fit_method`. Here, we use a two-stage approach: first, a tensor is fit to the data, and then this tensor as the initial starting point for the non-linear fit of IVIM parameters."?